spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/CompositePropagationFactory.java

```diff
@@ -124,7 +124,7 @@ Propagation.Factory map(PropagationType type) {
 		 * @return the B3 propagation factory
 		 */
 		private Propagation.Factory b3Single() {
-			return B3Propagation.newFactoryBuilder().injectFormat(B3Propagation.Format.SINGLE_NO_PARENT).build();
+			return B3Propagation.newFactoryBuilder().injectFormat(B3Propagation.Format.SINGLE).build();
```

Comment:
Should we add a test to verify this?
I think an extra verification step to one of our existing (integration?) tests is enough.

Reply:
Yep, i agree. Adding a test would be great.

Reply:
Ah that's a good idea, my bad! I won't have access to my laptop until next week and I can re-review.

Reply:
Hey both, I now have access to my laptop and am looking to review this. Quick question, should this test step be included in either of the existing unit tests in file CompositePropagationFactoryTests.java or should I create a new unit test?

Reply:
You can try to test this in `CompositePropagationFactoryTests` since you have access to the package-private `create` method, you might be able to test this in a very simple way. I think (didn't try) the tricky part will be the verification since you might not have access to the propagation format to check what it is.

What you can also try if this won't work is testing the behavior or the propagator. Check `BraveAutoConfigurationTests` there you can see how to create spans (you will need at least two: a parent and a child). You can create a test injector and "inject the span context into a variable". Then you can check what was injected. Before your change the parent id should not be there after that it should. Does this make sense? Please let us know if you need help.

Reply:
Hey @jonatan-ivanov thank you for the explanation, yes I think that makes sense. I will have a review and have a look into this and keep you updated :) have a lovely weekend.